docs-devsite/firestore_.firestoredataconverter.md

```diff
@@ -10,89 +10,91 @@ https://github.com/firebase/firebase-js-sdk
 {% endcomment %}
 
 # FirestoreDataConverter interface
-Converter used by `withConverter()` to transform user objects of type `T` into Firestore data.
+Converter used by `withConverter()` to transform user objects of type `AppModelType` into Firestore data of type `DbModelType`<!-- -->.
 
 Using the converter allows you to specify generic type arguments when storing and retrieving objects from Firestore.
 
 <b>Signature:</b>
 
 ```typescript
-export declare interface FirestoreDataConverter<T> 
+export declare interface FirestoreDataConverter<AppModelType, DbModelType extends DocumentData = DocumentData> 
 ```
 
 ## Methods
 
 |  Method | Description |
 |  --- | --- |
-|  [fromFirestore(snapshot, options)](./firestore_.firestoredataconverter.md#firestoredataconverterfromfirestore) | Called by the Firestore SDK to convert Firestore data into an object of type T. You can access your data by calling: <code>snapshot.data(options)</code>. |
-|  [toFirestore(modelObject)](./firestore_.firestoredataconverter.md#firestoredataconvertertofirestore) | Called by the Firestore SDK to convert a custom model object of type <code>T</code> into a plain JavaScript object (suitable for writing directly to the Firestore database). To use <code>set()</code> with <code>merge</code> and <code>mergeFields</code>, <code>toFirestore()</code> must be defined with <code>PartialWithFieldValue&lt;T&gt;</code>.<!-- -->The <code>WithFieldValue&lt;T&gt;</code> type extends <code>T</code> to also allow FieldValues such as [deleteField()](./firestore_.md#deletefield) to be used as property values. |
-|  [toFirestore(modelObject, options)](./firestore_.firestoredataconverter.md#firestoredataconvertertofirestore) | Called by the Firestore SDK to convert a custom model object of type <code>T</code> into a plain JavaScript object (suitable for writing directly to the Firestore database). Used with [setDoc()](./firestore_.md#setdoc)<!-- -->,  and  with <code>merge:true</code> or <code>mergeFields</code>.<!-- -->The <code>PartialWithFieldValue&lt;T&gt;</code> type extends <code>Partial&lt;T&gt;</code> to allow FieldValues such as [arrayUnion()](./firestore_.md#arrayunion) to be used as property values. It also supports nested <code>Partial</code> by allowing nested fields to be omitted. |
+|  [fromFirestore(snapshot, options)](./firestore_.firestoredataconverter.md#firestoredataconverterfromfirestore) | Called by the Firestore SDK to convert Firestore data into an object of type <code>AppModelType</code>. You can access your data by calling: <code>snapshot.data(options)</code>.<!-- -->Generally, the data returned from <code>snapshot.data()</code> can be cast to <code>DbModelType</code>; however, this is not guaranteed as writes to the database may have occurred without a type converter enforcing this specific layout. |
```

Comment:
Is there any way to clarify what cases the writes may occur without a type converter? It took me a second to realize this means that my application code may have spots that use a type converter and some that don't. In this sentence, you're referring to some of my application code that may not use a type converter. I think a less experienced customer could get hung up on this sentence and maybe choose not to cast to DbModelType when it's safe

Reply:
Sure, I'll update this documentation.

Reply:
Done in https://github.com/firebase/firebase-js-sdk/pull/7932